src/middleware/url.id.middleware.ts

```diff
@@ -0,0 +1,31 @@
+import { NextFunction, Request, Response } from "express";
+import { logger } from "../utils/logger";
+import { isUrlIdValid } from "../validators/url.id.validator";
+import { urlParams } from "../types/page.urls";
+
+export const urlIdMiddleware = (req: Request, res: Response, next: NextFunction) => {
+
+  // TODO This class needs to be unit-tested
+
+  logger.debug("Execute URL id middleware checks");
+
+  const transactionId: string = req.params[urlParams.PARAM_TRANSACTION_ID];
+
+  logger.debug("Check transaction id");
+  if (!isUrlIdValid(transactionId)) {
+    // TODO What should be done here and how? Entire parameter values appear (embedded) in the output URL but we truncate
```

Comment:
Example log output currently:

{"created":"2022-02-18T15:23:21.209+00:00","event":"debug","namespace":"confirmation-statement-web","data":{"message":"Check the URL id is valid"}}
{"created":"2022-02-18T15:23:21.210+00:00","event":"error","namespace":"confirmation-statement-web","data":{"message":"URL id exceeds 50 characters - 176113-2200AAAAAAAAAAAAAKJKJJKJKJKJKJKJKJKJKJAJKAJ..."}}
{"created":"2022-02-18T15:23:21.211+00:00","event":"error","namespace":"confirmation-statement-web","data":{"message":"No Valid Transaction Id in URL: /confirmation-statement/company/11456298/transaction/176113-2200AAAAAAAAAAAAAKJKJJKJKJKJKJKJKJKJKJAJKAJKJAKJKAJKJAKJKJAKJAKJA16-451977/submission/620fb9d4eba8327111a6ca41/sic","path":"/sic","method":"GET"}}
{"created":"2022-02-18T15:23:21.213+00:00","event":"error","namespace":"confirmation-statement-web","data":{"message":"An error has occurred. Re-routing to the error screen - Error: No Valid Transaction Id in Request\n    at urlIdMiddleware (/app/middleware/url.id.middleware.js:16:21)\n    at Layer.handle [as handle_request] (/app/node_modules/express/lib/router/layer.js:95:5)\n    at trim_prefix (/app/node_modules/express/lib/router/index.js:317:13)\n    a